File: examples/test-params.js
#!/usr/bin/env node
/**
 * Test Node.js script with parameters
 */

// Process argv access
console.log('Script:', process.argv[0]);
console.log('File:', process.argv[1]);
console.log('First arg:', process.argv[2]);
console.log('Second arg:', process.argv[3]);

// Simple argument parsing
const args = process.argv.slice(2);
let name = 'World';
let age = null;
let verbose = false;

for (let i = 0; i < args.length; i++) {
    if (args[i] === '--name' && args[i + 1]) {
        name = args[i + 1];
        i++; // skip next arg
    } else if (args[i] === '--age' && args[i + 1]) {
        age = parseInt(args[i + 1]);
        i++; // skip next arg
    } else if (args[i] === '--verbose') {
        verbose = true;
    }
}

console.log(`Hello ${name}!`);
if (age) {
    console.log(`You are ${age} years old`);
}
if (verbose) {
    console.log('Verbose mode enabled');
}

File: examples/test-params.py
#!/usr/bin/env python3
"""
Test script with various parameter patterns for analysis
"""
import sys
import argparse

def main():
    parser = argparse.ArgumentParser(description='Test script with parameters')
    parser.add_argument('--name', help='Your name')
    parser.add_argument('--age', type=int, help='Your age')
    parser.add_argument('--verbose', action='store_true', help='Enable verbose output')
    parser.add_argument('input_file', help='Input file path')
    
    args = parser.parse_args()
    
    print(f"Hello {args.name or 'World'}!")
    if args.age:
        print(f"You are {args.age} years old")
    if args.verbose:
        print("Verbose mode enabled")
    print(f"Processing file: {args.input_file}")
    
    # Also test sys.argv access
    if len(sys.argv) > 2:
        print(f"Second argument: {sys.argv[2]}")

if __name__ == "__main__":
    main()